torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -737,30 +827,58 @@ def auto_functionalized_func(ctx, _mutable_op, **kwargs):
 # auto_functionalized_v2 functions
 @auto_functionalized_v2.py_impl(DispatchKey.CompositeExplicitAutograd)
 def auto_functionalized_v2_dense(
-    _mutable_op: OpOverload,
+    _mutable_op: _MutableOpType,
     _only_clone_these_bases: Optional[tuple[int, ...]] = None,
     **kwargs: Any,
 ) -> tuple[Any, tuple[Tensor, ...]]:
     _all_bases: list[Tensor] = kwargs.pop("_all_bases", [])
     if _only_clone_these_bases is None:
         _only_clone_these_bases = tuple(range(len(_all_bases)))
 
-    schema = _mutable_op._schema
+    if isinstance(_mutable_op, OpOverload):
+        schema = _mutable_op._schema
+    else:
+        schema = pytree.tree_unflatten([], kwargs.pop("_op_schema")).schema
+
     op_kwargs_new, all_bases_new = _generate_new_op_kwargs_from_bases(
         schema,
         kwargs,
         _all_bases,
         _only_clone_these_bases,
     )
 
-    out = _mutable_op(**op_kwargs_new)
+    out = _invoke_op_with_kwargs_and_schema(
+        _mutable_op,
+        op_kwargs_new,
+        schema,
+    )
 
     if isinstance(out, tuple):
         return (*out, *all_bases_new)  # type: ignore[return-value]
     else:
         return (out, *all_bases_new)  # type: ignore[return-value]
 
 
+def _invoke_op_with_kwargs_and_schema(
```

Comment:
nit: maybe just "call_hop"? Can go into torch/_higher_order_ops.utils

I feel like we should split call_hop vs call_op. Because passing the schema for an opoverload doesn't make too much sense (we end up ignoring it).

e.g.

```
if isinstance(op, OpOverload):
    return op(**kwargs)
else:
    return call_hop(hop, schema, kwargs)
``` 
(hop, schema, kwargs) seems like the right order (schema is tied to the hop)

Reply:
I see you did do the HopInstance thing. Alternative, this can just be:
def call_op(op: Union[OpOverload, HOPInstance], args, kwargs):
    if isinstance(op, OpOverload):
        return op(*args, **kwargs)
    # HOPInstance case: pull schema from it
    schema = op._schema
    ...